xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java

```diff
@@ -469,6 +478,33 @@ public void run() {
         new EndpointsUpdated().run();
       }
 
+      private SocketAddress rewriteAddress(SocketAddress addr, ImmutableMap<String,
+          Object> endpointMetadata, ImmutableMap<String, Object> localityMetadata) {
+        if (!(addr instanceof InetSocketAddress)) {
+          return addr;
+        }
+
+        SocketAddress proxyAddress;
+        try {
+          proxyAddress = (SocketAddress) endpointMetadata.get(
+              "envoy.http11_proxy_transport_socket.proxy_address");
+          if (proxyAddress == null) {
+            proxyAddress = (SocketAddress) localityMetadata.get(
+                "envoy.http11_proxy_transport_socket.proxy_address");
+          }
+        } catch (Exception e) {
+          return addr;
+        }
+
+        if (proxyAddress != null) {
```

Comment:
For consistency with the rest of this method, invert this to `if (proxyAddress == null) return addr;`?